### PR TITLE
Dtspo 5032 ptlprod aks

### PR DESCRIPTION
**Pull request by darren221980**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/DTSPO-5032

### Change description ###

Updated pipeline to include PTL env, added tfvars for PTL for Network & Network RG

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X ] No
```

### CODE REVIEW

File: components/managed-identity/inputs-default.tf [hunk 1]
```diff
@@ -11,7 +11,7 @@ locals {
   log_analytics_env_mapping = {
     sandbox = ["sbox", "ptlsbox"]
     nonprod = ["dev", "perftest", "ithc", "demo", "aat"]
-    prod    = ["prod", "mgmt"]
+    prod    = ["prod", "mgmt", "ptl"]
```
Comment (thomast1906) [inline components/managed-identity/inputs-default.tf:14]: Is this mapping correct?

File: components/managed-identity/interpolated-defaults.tf [hunk 2]
```diff
@@ -10,7 +10,7 @@ data "azurerm_resource_group" "genesis_rg" {
 
 locals {
 
-  environment = (var.environment == "perftest") ? "test" : (var.environment == "aat") ? "stg" : (var.environment == "ptlsbox") ? "sbox" : "${var.environment}"
+  environment = (var.environment == "perftest") ? "test" : (var.environment == "aat") ? "stg" : (var.environment == "ptlsbox") ? "sbox" : (var.environment == "ptl") ? "prod" : "${var.environment}"
```
Comment (thomast1906) [inline components/managed-identity/interpolated-defaults.tf:13]: Why is it being changed to prod?

File: environments/network/ptl.tfvars [hunk 3]
```diff
@@ -0,0 +1,107 @@
+enable_debug = "true"
+
+network_address_space                  = "10.10.72.0/21 "
+aks_00_subnet_cidr_blocks              = "10.10.72.0/23"
+aks_01_subnet_cidr_blocks              = "10.10.74.0/23"
+iaas_subnet_cidr_blocks                = "10.10.78.0/23"
+application_gateway_subnet_cidr_blocks = "10.10.70.0/25"
```
Comment (thomast1906) [inline environments/network/ptl.tfvars:7]: Not correct to the `network_address_space`

File: environments/network/ptl.tfvars [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+  "service.core-compute-idam-prod2.internal",
+  "service.core-compute-ithc.internal",
+  "service.core-compute-perftest.internal",
+  "service.core-compute-preview.internal",
+  "service.core-compute-prod.internal",
+  "staging.platform.hmcts.net",
+  "test.platform.hmcst.net",
+]
+
+hub = "prod"
```
Comment (thomast1906) [inline environments/network/ptl.tfvars:54]: Production Jenkins needs to peer to both prod & non-prod hubs

File: pipeline-steps/deploy-network.yaml [hunk 5]
```diff
@@ -36,7 +36,9 @@ steps:
          backendAzureRmResourceGroupName: 'azure-control-stg-rg'
       ${{ if eq( parameters['environment'], 'ptlsbox') }}: 
          backendAzureRmResourceGroupName: 'azure-control-sbox-rg'
-      ${{ if notIn( parameters['environment'], 'aat', 'perftest', 'ptlsbox') }}: 
+      ${{ if eq( parameters['environment'], 'ptl') }}: 
+         backendAzureRmResourceGroupName: 'azure-control-prod-rg'
```
Comment (thomast1906) [inline pipeline-steps/deploy-network.yaml:40]: Is the `azure-control-prod-rg` references correct?

File: components/managed-identity/interpolated-defaults.tf (hunk 2 above)
Comment (darren221980) [reply components/managed-identity/interpolated-defaults.tf:13]: This is changed due to the KV that the managed identity is calling from, being in the azure-control-prod-rg RG.

File: components/managed-identity/inputs-default.tf (hunk 1 above)
Comment (darren221980) [reply components/managed-identity/inputs-default.tf:14]: Existing PTL environment is sending to the hmcts-prod log analytics workspace.

File: environments/network/ptl.tfvars (hunk 3 above)
Comment (darren221980) [reply environments/network/ptl.tfvars]: all updated

File: environments/network/ptl.tfvars [hunk 6]
```diff
@@ -0,0 +1,107 @@
+enable_debug = "true"
+
+network_address_space                  = "10.10.72.0/21 "
+aks_00_subnet_cidr_blocks              = "10.10.72.0/23"
+aks_01_subnet_cidr_blocks              = "10.10.74.0/23"
+iaas_subnet_cidr_blocks                = "10.10.78.0/23"
```
[the remark this answers is not in the document]
Comment (darren221980) [reply environments/network/ptl.tfvars]: all updated

File: pipeline-steps/deploy-network.yaml (hunk 5 above)
Comment (darren221980) [reply pipeline-steps/deploy-network.yaml:40]: yes.

File: components/network-rg/network-rg.tf [hunk 7]
```diff
@@ -8,6 +8,15 @@ resource "azurerm_resource_group" "network_resource_group" {
   tags = module.ctags.common_tags
 }
 
+resource "azurerm_resource_group" "core_infra_resource_group" {
```
Comment (thomast1906) [inline components/network-rg/network-rg.tf:11]: Why is a new resource added here?

File: environments/network/ptl.tfvars [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+  },
+  {
+    name                   = "cftithc"
+    address_prefix         = "10.11.192.0/18"
+    next_hop_type          = "VirtualAppliance"
+    next_hop_in_ip_address = "10.11.72.36"
+  },
+]
+
+additional_routes_coreinfra = [
```
Comment (thomast1906) [inline environments/network/ptl.tfvars:82]: Are these routes needed?

File: environments/network/ptl.tfvars [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+  "service.core-compute-perftest.internal",
+  "service.core-compute-preview.internal",
+  "service.core-compute-prod.internal",
+  "staging.platform.hmcts.net",
+  "test.platform.hmcts.net",
+]
+
+hub = "prod"
+
+additional_routes = [
```
Comment (thomast1906) [inline environments/network/ptl.tfvars:49]: Missing the standard RFC1918 routes?

Comment (darren221980) [reply environments/network/ptl.tfvars:49]: Now added.

File: environments/aks/sbox.tfvars [hunk 10]
```diff
@@ -3,9 +3,9 @@ kubernetes_cluster_version         = "1.19.11"
 kubernetes_cluster_agent_min_count = "6"
 kubernetes_cluster_agent_max_count = "15"
 kubernetes_cluster_ssh_key         = "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQCeRwSzSKJfjmIVQ6CUld/M3vF9Hcfxh5cLBa1BV+UZDh5p1gKoB0xRegSFdncfup1qMAhrZtgBpaclLiYUfe8ZajPp1Lmva9AJuK/UktzF9stZie7LDpflEdVBXlSZw3AtAWxF2vIkEeW+NVYlGAJOQlasFkmGTkco+O1wUM4LGI3YNHm7r70rnmHT2djoR1t4x1jlPCrXaJEhvtyxf01Dwjq2nWaox3puJtHs5DLFpEIvXvHwQWssFFyKIuwkm4FewHKJSbCahyaCb+ac10MAZg9vZnWq0EYOe1nLn7c3538yJ9WJh7jRFZDza6ab9HVb5hgJ3/t/K+EzkU/XGSEJ"
-enable_user_system_nodepool_split = true
-project_acr_enabled               = true
-sku_tier                          = "Paid"
+enable_user_system_nodepool_split  = true
```
Comment (thomast1906) [inline environments/aks/sbox.tfvars:6]: Why is sbox.tfvars in your PR?

Comment (thomast1906) [reply environments/aks/sbox.tfvars:6]: actually, looks like fmt only

**Review by thomast1906 — APPROVED** [review]

**Pull request merged**